test/LanguageServer.Engine.Tests/MSBuildObjectLocatorTests.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Build.Evaluation;
+using MSBuildProjectTools.LanguageServer.SemanticModel;
+using System;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MSBuildProjectTools.LanguageServer.Tests
+{
+    /// <summary>
+    ///     Tests for locating MSBuild objects by position.
+    /// </summary>
+    [Collection(MSBuildEngineFixture.CollectionName)]
+    public class MSBuildObjectLocatorTests
+        : TestBase, IDisposable
+    {
+        /// <summary>
+        ///     The directory for test files.
+        /// </summary>
+        static readonly DirectoryInfo TestDirectory = new DirectoryInfo(Path.GetDirectoryName(
+            typeof(MSBuildObjectLocatorTests).Assembly.Location
+        ));
+
+        /// <summary>
+        ///     The project collection for any projects loaded by the current test.
+        /// </summary>
+        ProjectCollection _projectCollection;
```

Comment:
I am very uncomfortable having field which represents per-test data. Yes, it happens that XUnit runs tests in the same collection sequentially, but this is still a recipe for concurrency disaster if it ever changes or we annotate different tests in the file to be in different collections (that is possible, right?) and so on. You should create project collection in each test and use `using` statement, so it can be disposed properly. That would also remove the need to care about `null`s, implementing `IDisposable` and so on

Reply:
This was meant to be the first phase of producing base classes for test suites that work with XML, projects, etc (not that I’d indicated as such but it’s what I had in mind). I’d rather not have verbose setup code for each test (especially if they are the same) but I’ll have a think about whether we can make a little wrapper for it that can be used as you describe (I just want to avoid the situation where each test has to get the setup exactly right and it becomes hard to determine whether 2 tests are really creating the same configuration, besides the bits that are actually mean to be different from test to test)…

Reply:
> I am very uncomfortable having field which represents per-test data

[As it turns out](https://xunit.net/docs/shared-context), xUnit creates a new instance of the test class per test:

> xUnit.net creates a new instance of the test class for every test that is run

so using fields like this seems to be fine.

Reply:
> so using fields like this seems to be fine.

Yes indeed, that would work. But from pure design perspective having fields for per-test data should be avoided unless it is absolutely needed. Because now you depend on XUnit behavior and make everyone else contributing to actively remember that fact to be sure, that the field's value won't change during each test case run. Given, that this PR is merged, I think it is no longer relevant, but I would urge to refrain from this pattern in the future.